sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HttpHealthcareApiClient.java

```diff
@@ -0,0 +1,459 @@
+        .create(hl7v2Store, createMessageRequest)
+        .execute();
+  }
+
+  @Override
+  public HttpBody createFhirResource(String fhirStore, String type, HttpBody body)
+      throws IOException {
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .fhirStores()
+        .fhir()
+        .create(fhirStore, type, body)
+        .execute();
+  }
+
+  @Override
+  public HttpBody executeFhirBundle(String fhirStore, HttpBody bundle) throws IOException {
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .fhirStores()
+        .fhir()
+        .executeBundle(fhirStore, bundle)
+        .execute();
+  }
+
+  @Override
+  public HttpBody listFHIRResourceForPatient(String fhirStore, String patient) throws IOException {
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .fhirStores()
+        .fhir()
+        .patientEverything(patient)
+        .execute();
+  }
+
+  @Override
+  public HttpBody readFHIRResource(String fhirStore, String resource) throws IOException {
+    return client.projects().locations().datasets().fhirStores().fhir().read(resource).execute();
+  }
+
+  private static class AuthenticatedRetryInitializer extends RetryHttpRequestInitializer {
+    GoogleCredentials credentials;
+
+    public AuthenticatedRetryInitializer(GoogleCredentials credentials) {
+      super();
+      this.credentials = credentials;
+    }
+
+    @Override
+    public void initialize(HttpRequest request) throws IOException {
+      super.initialize(request);
+      if (!credentials.hasRequestMetadata()) {
+        return;
+      }
+      HttpHeaders requestHeaders = request.getHeaders();
+      requestHeaders.setUserAgent("apache-beam-hl7v2-io");
+      URI uri = null;
+      if (request.getUrl() != null) {
+        uri = request.getUrl().toURI();
+      }
+      Map<String, List<String>> credentialHeaders = credentials.getRequestMetadata(uri);
+      if (credentialHeaders == null) {
+        return;
+      }
+      for (Map.Entry<String, List<String>> entry : credentialHeaders.entrySet()) {
+        String headerName = entry.getKey();
+        List<String> requestValues = new ArrayList<>(entry.getValue());
+        requestHeaders.put(headerName, requestValues);
+      }
+    }
+  }
+
+  private void initClient() throws IOException {
+    // Create a HttpRequestInitializer, which will provide a baseline configuration to all requests.
+    // HttpRequestInitializer requestInitializer = new RetryHttpRequestInitializer();
+    // GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+    HttpRequestInitializer requestInitializer =
+        new AuthenticatedRetryInitializer(
+            GoogleCredentials.getApplicationDefault()
+                .createScoped(CloudHealthcareScopes.CLOUD_PLATFORM));
+
+    client =
+        new CloudHealthcare.Builder(new NetHttpTransport(), new GsonFactory(), requestInitializer)
```

Comment:
Memo: this uses `com.google.api.client.json.gson.GsonFactory`.

Reply:
I see CloudHealthcare.Builder accepts JacksonFactory.
https://github.com/googleapis/google-api-java-client-services/blob/master/clients/google-api-services-healthcare/v1beta1/1.30.1/com/google/api/services/healthcare/v1beta1/CloudHealthcare.java#L16349